homeassistant/components/assist_pipeline/pipeline.py

```diff
@@ -393,6 +422,121 @@ def end(self) -> None:
+                sample_rate=stt.AudioSampleRates.SAMPLERATE_16000,
+                channel=stt.AudioChannels.CHANNEL_MONO,
+            )
+        )
+
+        # Remove language since it doesn't apply to wake words yet
+        metadata_dict.pop("language", None)
+
+        self.process_event(
+            PipelineEvent(
+                PipelineEventType.WAKE_START,
+                {
+                    "engine": self.wake_engine,
+                    "metadata": metadata_dict,
+                },
+            )
+        )
+
+        wake_settings = self.wake_settings or WakeSettings()
+
+        wake_vad: VoiceActivityTimeout | None = None
+        if (wake_settings.timeout is not None) and (wake_settings.timeout > 0):
+            # Use VAD to determine timeout
+            wake_vad = VoiceActivityTimeout(wake_settings.timeout)
+
+        # Audio chunk buffer.
+        # Keeping audio right before wake word detection allows the voice
+        # command to be spoken immediately after the wake word.
+        chunk_buffer: deque[bytes] | None = None
+        if wake_settings.audio_chunks_to_buffer > 0:
+            chunk_buffer = deque(maxlen=wake_settings.audio_chunks_to_buffer)
+
+        async def timestamped_stream() -> AsyncIterable[tuple[bytes, int]]:
+            """Yield audio with timestamps (milliseconds since start of stream)."""
+            timestamp_ms = 0
+            async for chunk in stream:
+                yield chunk, timestamp_ms
+                timestamp_ms += (len(chunk) // 2) // 16  # milliseconds @ 16Khz
+
+                if chunk_buffer is not None:
+                    chunk_buffer.append(chunk)
+
+                if (wake_vad is not None) and (not wake_vad.process(chunk)):
+                    raise WakeWordTimeoutError(
+                        code="wake-word-timeout", message="Wake word was not detected"
+                    )
+
+        try:
+            # Detect wake word(s)
+            result = await self.wake_provider.async_process_audio_stream(
+                timestamped_stream()
+            )
+        except WakeWordTimeoutError:
+            _LOGGER.debug("Timeout during wake word detection")
+            raise
+        except Exception as src_error:
+            _LOGGER.exception("Unexpected error during wake-word-detection")
+            raise WakeWordDetectionError(
+                code="wake-stream-failed",
+                message="Unexpected error during wake-word-detection",
+            ) from src_error
+
+        _LOGGER.debug("wake-word-detection result %s", result)
+
+        if result is None:
+            wake_output: dict[str, Any] = {}
+        else:
+            if chunk_buffer:
+                audio_buffer.extend(chunk_buffer)
+
+            wake_output = asdict(result)
+
+        self.process_event(
+            PipelineEvent(
+                PipelineEventType.WAKE_END,
+                {"wake_output": wake_output},
+            )
+        )
```

Comment:
I am still receiving this event with no result instead of an error

```
2023-07-27 20:01:42.050 DEBUG (MainThread) [homeassistant.components.assist_pipeline.pipeline] wake-word-detection result None
```

Reply:
I've pushed all of my latest changes, including renaming the events to `wake_word-*`
I'm hoping your problem will be magically fixed :crossed_fingers: